blobmodel/model.py

```diff
@@ -180,13 +189,30 @@ def _sum_up_blobs(
             periodic_y=self._geometry.periodic_y,
             Ly=self._geometry.Ly,
         )
-        self._density[:, :, _start:_stop] += _single_blob
+
         if self._labels:
-            _max_amplitudes = np.max(_single_blob, axis=(0, 1))
-            _max_amplitudes[_max_amplitudes == 0] = np.inf
-            self._labels_field[:, :, _start:_stop][
-                _single_blob >= _max_amplitudes * self._label_border
-            ] = 1
+            if self.label_type == "simple":
+                __max_amplitudes = np.max(_single_blob, axis=(0, 1))
+                __max_amplitudes[__max_amplitudes == 0] = np.inf
+                self._labels_field[:, :, _start:_stop][
+                    _single_blob >= __max_amplitudes * self._label_border
+                ] = 1
+            if self.label_type == "simple diff":
+                __max_amplitudes = np.max(_single_blob, axis=(0, 1))
+                __max_amplitudes[__max_amplitudes == 0] = np.inf
+                self._labels_field[:, :, _start:_stop][
+                    _single_blob >= __max_amplitudes * self._label_border
+                ] = (blob.blob_id + 1)
+
+            if self.label_type == "diff":
+                __max_amplitudes = np.max(_single_blob, axis=(0, 1))
+                __max_amplitudes[__max_amplitudes == 0] = np.inf
+                self._labels_field[:, :, _start:_stop][
+                    _single_blob
+                    >= __max_amplitudes * self._label_border
+                    + self._density[:, :, _start:_stop] * 0.25
```

Comment:
I don't understand the point of this line, please provide an explanation of the different options of `label_type`